### DISCUSSION TITLE
Publish as NPM Package

### DISCUSSION THREAD

**Discussion by MCatherine1994** [bcgov/nr-theme#12] [Ideas]
I wonder if there is any restriction about publishing our FSA theme stylesheet as an NPM package in the public node package registry? We'll need to [create an account](https://docs.npmjs.com/creating-a-new-npm-user-account) in the node package registry. Public registry is free, private one costs $7 per month. Or is there anywhere else we could publish, any GitHub registry for our organization?

Test it locally:
- Clone the nr-fsa-theme repo, go to its folder, inside the root directory, run `npm init` to initialize as a node package
- In another project, run `npm install [path_to_our_fsa_theme_repo]`
- And import the stylesheet as `import "nr-fsa-theme/theme/primevue-components-overrides.scss"`

How other teams use it in the future:
- install our nr-fsa-theme as a package
- whenever there is an update happen in the common repo, we publish a new version for the common repo, and other teams just install the latest the version

An example: https://github.com/GovAlta/design-tokens
NRDK npm repo: https://github.com/BCDevOps/nrdk

@DerekRoberts @webgismd @mamartinezmejia

**Comment by DerekRoberts** [comment]
This is a great idea! Nrdk used to do something similar, so we can check that repo for samples or ideas.

**Comment by MCatherine1994** [comment]
Derek created an organization in the npm registry for us: https://www.npmjs.com/settings/bcgov-nr/members

**Comment by MCatherine1994** [comment]
Manual publish steps:
- `npm adduser`
- `npm publish --access public`

![image](https://user-images.githubusercontent.com/23131735/231315764-7176260e-590f-44f7-8bab-0e687b69397d.png)
